plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -1301,6 +1306,28 @@ int create_log_stream(struct flb_cloudwatch *ctx, struct log_stream *stream)
                     flb_http_client_destroy(c);
                     return 0;
                 }
+
+                if (strcmp(error, ERR_CODE_NOT_FOUND) == 0) {
+                    flb_sds_destroy(body);
+                    flb_sds_destroy(error);
+                    flb_http_client_destroy(c);
+
+                    if (ctx->create_group == FLB_TRUE) {
+                        flb_plg_info(ctx->ins, "Log Group %s not found. Will attempt to create it.",
+                                 ctx->log_group);
+                        ret = create_log_group(ctx);
+                        if (ret < 0) {
+                            return -1;
+                        } else {
+                            /* retry stream creation */
+                            goto retry_create_stream;
```

Comment:
Should there be a retry limit? This could be an infinite loop if the resource continues to fail.

Reply:
What is the advantage of using goto statement here vs `return create_log_stream(ctx, stream);`? Is it for efficiency so the function stack doesn't need to be reinitialized?

Reply:
>  Because the call to create_stream only happens again if the CreateLogGroup succeeds but then create stream again fails with the resource not found exception. Which shouldn't be able to happen repeatedly.

Just a suggestion - I think it might be better to add a bool here so that we are sure that the creation won't happen again and again. It might be very low possibility but just in case.